modyn/evaluator/internal/grpc/evaluator_grpc_servicer.py

```diff
@@ -0,0 +1,223 @@
+    def evaluate_model(self, request: EvaluateModelRequest, context: grpc.ServicerContext) -> EvaluateModelResponse:
+        logger.info("Received evaluate model request.")
+
+        if not hasattr(dynamic_module_import("modyn.models"), request.model_id):
+            logger.error(f"Model {request.model_id} not available!")
+            return EvaluateModelResponse(evaluation_started=False)
+
+        fetch_request = FetchModelRequest(model_id=request.trained_model_id)
+        fetch_resp: FetchModelResponse = self._model_storage_stub.FetchModel(fetch_request)
+
+        if not fetch_resp.success:
+            logger.error(
+                f"Trained model {request.trained_model_id} cannot be fetched from model storage. "
+                f"Evaluation cannot be started."
+            )
+            return EvaluateModelResponse(evaluation_started=False)
+
+        with self._lock:
+            evaluation_id = self._next_evaluation_id
+            self._next_evaluation_id += 1
+
+        local_model_path = self._base_dir / f"trained_model_{evaluation_id}.modyn"
+
+        download_file(
+            hostname=self._config["model_storage"]["hostname"],
+            port=int(self._config["model_storage"]["ftp_port"]),
+            user="modyn",
+            password="modyn",
+            remote_file_path=pathlib.Path(fetch_resp.model_path),
+            local_file_path=local_model_path,
+            callback=get_pretrained_model_callback(logger),
+        )
+
+        logger.info(f"Successfully downloaded trained model {request.trained_model_id} to {local_model_path}.")
+
+        metrics = self._setup_metrics(request.metrics)
+        evaluation_info = EvaluationInfo(request, evaluation_id, self._storage_address, metrics, local_model_path)
+        self._evaluation_dict[evaluation_id] = evaluation_info
+
+        exception_queue: mp.Queue[str] = mp.Queue()  # pylint: disable=unsubscriptable-object
+        status_query_queue: mp.Queue[str] = mp.Queue()  # pylint: disable=unsubscriptable-object
+        status_response_queue: mp.Queue[dict[str, Any]] = mp.Queue()  # pylint: disable=unsubscriptable-object
+
+        process = mp.Process(
+            target=evaluate,
+            args=(
+                self._evaluation_dict[evaluation_id],
+                self._base_dir / f"log-{evaluation_id}.txt",
+                exception_queue,
+                status_query_queue,
+                status_response_queue,
+            ),
+        )
+        process.start()
+        self._evaluation_process_dict[evaluation_id] = EvaluationProcessInfo(
+            process, exception_queue, status_query_queue, status_response_queue
+        )
+
+        logger.info(f"Started evaluation {evaluation_id}.")
+        return EvaluateModelResponse(evaluation_started=True, evaluation_id=evaluation_id)
+
+    def _setup_metrics(self, metrics: list[str]) -> list[AbstractEvaluationMetric]:
+        return list(map(self._metric_manager.get_evaluation_metric, set(metrics)))
```

Comment:
could you write a comment what this is doing? sorry, just from looking at it it's not clear to me why we need it